src/cargo/core/compiler/custom_build.rs

```diff
@@ -643,6 +643,15 @@ impl BuildOutput {
                     }
                     linker_args.push((LinkType::SingleBin(bin_name), arg.to_string()));
                 }
+                "rustc-link-arg-tests" => {
+                    linker_args.push((LinkType::Test, value));
```

Comment:
Can these include validation that the package actually has a test or bench or example?  Similar to `rustc-link-arg-bins`, this is intended to help the developer know that the instruction isn't valid.  And if you can share the code between the bins/tests/benches/examples cases, that would be great!